nvflare/app_common/utils/fl_model_utils.py

```diff
@@ -104,27 +104,33 @@ def from_shareable(
         In the future, we should be using the to_dxo, from_dxo directly.
         And all the components should be changed to accept the standard DXO.
         """
-        dxo = from_shareable(shareable)
         metrics = None
         params_type = None
         params = None
+        meta = {}
 
-        if dxo.data_kind == DataKind.METRICS:
-            metrics = dxo.data
-        else:
-            params_type = data_kind_to_params_type.get(dxo.data_kind)
-            if params_type is None:
-                raise ValueError(f"Invalid shareable with dxo that has data kind: {dxo.data_kind}")
-            params_type = ParamsType(params_type)
-            if params_converter:
-                dxo.data = params_converter.convert(dxo.data)
-            params = dxo.data
+        try:
+            dxo = from_shareable(shareable)
+            meta = dict(dxo.meta)
+            if dxo.data_kind == DataKind.METRICS:
+                metrics = dxo.data
+            else:
+                params_type = data_kind_to_params_type.get(dxo.data_kind)
+                if params_type is None:
+                    raise ValueError(f"Invalid shareable with dxo that has data kind: {dxo.data_kind}")
+                params_type = ParamsType(params_type)
+                if params_converter:
+                    dxo.data = params_converter.convert(dxo.data)
+                params = dxo.data
+        except:
+            # this only happens in cross-site eval right now
+            submit_model_name = shareable.get_header(AppConstants.SUBMIT_MODEL_NAME)
+            meta[MetaKey.SUBMIT_MODEL_NAME] = submit_model_name
```

Comment:
I see this submit_model_name is set in the meta, is this used anywhere in cse?

Reply:
```
./nvflare/app_common/executors/model_learner_executor.py:        model_name = shareable.get_header(AppConstants.SUBMIT_MODEL_NAME)
./nvflare/app_common/executors/learner_executor.py:        model_name = shareable.get_header(AppConstants.SUBMIT_MODEL_NAME)
./nvflare/app_common/ccwf/cse_client_ctl.py:            task_data.set_header(AppConstants.SUBMIT_MODEL_NAME, model_name)
```

Reply:
so to integrate using client api, on the 3rd party scripts we need to use meta to get this model name